### PR TITLE
feat: allow to filter branches

### PR DESCRIPTION
**Pull request by KillianHmyd**
This can be useful when users have multiple branches from where they trigger a release.
Having this filter would allow to only send a changelog from master branches for example.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -93,6 +93,7 @@ Alternatively, you could pass the webhook as a configuration option.
...
 | `packageName`          | Override or add package name instead of npm package name                                                                                                                                                                                                                                          | SEMANTIC_RELEASE_PACKAGE or npm package name                   |
 | `unsafeMaxLength`      | Maximum character length for the release notes before truncation. If unsafeMaxLength is too high, messages can be dropped. [Read here](https://github.com/juliuscc/semantic-release-slack-bot/issues/26#issuecomment-569804359) for more information. Set to '0' to turn off truncation entirely. | 2900                                                           |
+| `filterBranches`       | Avoid triggering a slack message to be sent on branches not mathcing these filters. If not defined, slack message will be triggered on every branches. Filters can be glob patterns.                                                            | undefined                                    |
```
Comment (0xCAFEADD1C7) [inline README.md:96]:
```suggestion
| `filterBranches`       | Avoid triggering a slack message to be sent on branches not matching these filters. If not defined, slack message will be triggered on every branches. Filters can be glob patterns.                                                            | undefined                                    |
```

Comment (KillianHmyd) [reply README.md:96]: Thanks 👍

**Comment by juliuscc** [comment]
I love this idea! I just looked through it very quickly but will give it a more thorough review later 👍 . Could you add this new property to the example in the readme?

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,9 @@
+FROM alpine/git:v2.30.0
+
+RUN mkdir /ghcli
+WORKDIR /ghcli
+RUN wget https://github.com/cli/cli/releases/download/v1.5.0/gh_1.5.0_linux_386.tar.gz -O ghcli.tar.gz
+RUN tar --strip-components=1 -xf ghcli.tar.gz
+RUN mv bin/gh /bin/
+
+ENTRYPOINT ["/bin/ash"]
```
Comment (juliuscc) [inline Dockerfile:9]: What does this do?

Comment (KillianHmyd) [reply Dockerfile]: Sorry for that, this should not be here at all ! I've removed it

**Comment by KillianHmyd** [comment]
We talked about this proposal with my colleagues and we think we can add a "smarter" feature than that.
Instead of having a field `filterBranches` which allow to omit the slack notification on specific branches we could add an option to specific custom config according to the branches, so for example we could have a config like that to only send a slack message on master: 
```json
[
      "semantic-release-slack-bot",
      {
       "notifyOnSuccess": false,
        "notifyOnFail": false,
        "branchesConfig": [{
           "branchPattern": 'master',
           "notifyOnSuccess": true,
           "notifyOnFail": true,
        }]
      }
    ]
``` 

The advantage of that would be to specify different channel according to the branches : 
```json
[
      "semantic-release-slack-bot",
      {
       "notifyOnSuccess": false,
        "notifyOnFail": false,
        "branchesConfig": [
           {
             "branchPattern": 'master',
             "notifyOnSuccess": true,
             "notifyOnFail": true,
             "slackWebhook": "stable-channel",
          },
           {
             "branchPattern": 'lts/x',
             "notifyOnSuccess": true,
             "notifyOnFail": true,
             "slackWebhook": "lts-channel",
          }
       ]
      }
    ]
``` 

What do you think about it ? We can make a proposal in another PR if you want 👍

**Comment by juliuscc** [comment]
I think that is a much better idea! I assume that is exactly what PR #63 does? If so let's close this and continue at the other PR 👍

**Pull request closed without merging**